test/passes/optimize-instructions_fuzz-exec.txt

```diff
@@ -321,13 +324,35 @@
    )
   )
  )
+ (func $shift (param $0 i32)
+  (call $log
+   (i32.shr_s
+    (i32.shl
+     (i32.shr_s
+      (i32.shl
+       (local.get $0)
+       (i32.const 24)
+      )
+      (i32.const 24)
+     )
+     (i32.const 30)
+    )
+    (i32.const 24)
```

Comment:
Wouldn't it be possible to turn this into a left shift by 6?

Reply:
No, because of the possible effective sign bit. Consider this:
```wat
00000100000
;; shift left by x+y (x, y > 0)
10000000000
;; signed shift right by y
11100000000

;; start again
00000100000
;; shift left by just x and that's it:
00100000000
```